src/openMVG/cameras/Camera_Pinhole.hpp

```diff
@@ -108,11 +108,13 @@ class Pinhole_Intrinsic : public IntrinsicBase
   // Data wrapper for non linear optimization (update from data)
   virtual bool updateFromParams(const std::vector<double> & params)
   {
-    if (params.size() == 3) {
-      *this = Pinhole_Intrinsic(_w, _h, params[0], params[1], params[2]);
+    if (params.size() == 3)
+    {
+      this->setK(params[0], params[1], params[2]);
```

Comment:
dunno, isn't that we are replicating code here, instead of calling the constructor?
I mean, if for some reason we were to add another parameter one day we need to change the constructor AND here. But I don't know, it's true that it will be not likely the case ...